### PR TITLE
P2582R1 Wording for class template argument deduction from inherited …

### LINKED ISSUE #5581: [2022-07 CWG Motion 16] P2582R1 Wording for class template argument deduction from inherited constructors
[P2582R1](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2582r1.pdf)

### PR DESCRIPTION
**Pull request by burblebee**
…constructors

Fixes #5581
Fixes cplusplus/papers#1242

### CODE REVIEW

**Comment by JohelEGP** [comment]
Also fixes the corresponding issue at https://github.com/cplusplus/papers/.

File: source/overloading.tex [hunk 1]
```diff
@@ -1229,6 +1229,69 @@
...
+\begin{codeblock}
+template <typename T> struct B {
+  B(T);
+};
+template <typename T> struct C : public B<T> {
+  using B<T>::B;
+};
+template <typename T> struct D : public B<T> {};
+
+C c(42);            // OK, \tcode{deduces C<int>}
```
Comment (JohelEGP) [inline source/overloading.tex:1274]:
```suggestion
C c(42);            // OK, deduces \tcode{C<int>}
```

File: source/overloading.tex [hunk 2]
```diff
@@ -1229,6 +1229,69 @@
...
+D d(42);            // Error: deduction failed, no inherited deduction guides
```
Comment (JohelEGP) [inline source/overloading.tex:1275]:
```suggestion
D d(42);            // error: deduction failed, no inherited deduction guides
```

File: source/overloading.tex [hunk 3]
```diff
@@ -1229,6 +1229,69 @@
...
+B(int) -> B<char>;
+C c2(42);           // OK, deduces \tcode{C<char>}
+
+template <typename T> struct E : public B<int> {
+  using B<int>::B;
+};
+
+E e(42);            // Error: deduction failed, arguments of \tcode{E} cannot be deduced from guides introduced
```
Comment (JohelEGP) [inline source/overloading.tex:1283]:
```suggestion
E e(42);            // error: deduction failed, arguments of \tcode{E} cannot be deduced from guides introduced
```
```suggestion
E e(42);            // error: deduction failed, arguments of \tcode{E} cannot be deduced from introduced guides
```

File: source/overloading.tex [hunk 4]
```diff
@@ -1697,6 +1760,18 @@
...
+from class template argument deduction\iref{over.match.class.deduct}
+for a class \tcode{D}, and
+\tcode{F2} is generated
+from inheriting constructors from a base class of \tcode{D}
+while \tcode{F1} is not, and
+for each explicit function argument,
+the corresponding parameters of \tcode{F1} and \tcode{F2}
+are either both ellipses or have the same type,
+or, if not that,
+
```
Comment (JohelEGP) [inline source/overloading.tex:1774]: This seems to be inserted between the wrong items.

**Review by JohelEGP — CHANGES_REQUESTED** [review]

File: source/overloading.tex [hunk 5]
```diff
@@ -1229,6 +1229,69 @@
...
+replaced with \tcode{typename CC<R>::type} given a class template
+\begin{codeblock}
+template <typename> class CC;
+\end{codeblock}
+whose primary template is not defined and
+with a single partial specialization
+whose template parameter list is that of \tcode{A} and
+whose template argument list is a specialization of \tcode{A} with
+the template argument list of \tcode{A}\iref{temp.dep.type}
+having a member typedef type designating a template specialization with
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/overloading.tex:1298]: fixed

File: source/overloading.tex (hunk 1 above)
Comment (burblebee) [reply source/overloading.tex:1319]: doh, fixed

File: source/overloading.tex (hunk 2 above)
Comment (burblebee) [reply source/overloading.tex:1320]: fixed

File: source/overloading.tex (hunk 4 above)
Comment (burblebee) [reply source/overloading.tex:1819]: fixed, thanks!

File: source/overloading.tex (hunk 3 above)
Comment (burblebee) [reply source/overloading.tex:1328]: fixed

**Pull request merged**